app/workers/backend_delete_application_worker.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+class BackendDeleteApplicationWorker < ApplicationJob
+
+  queue_as :low
+
+  def self.enqueue(event)
+    perform_later(event.event_id)
+  end
+
+  def perform(event_id)
+    event = EventStore::Repository.find_event!(event_id)
+
+    @application = build_application(event)
+
+    [ApplicationKey, ReferrerFilter].each(&method(:delete_backend_associations))
+
+    delete_backend_application
+  rescue ActiveRecord::RecordNotFound => exception
+    System::ErrorReporting.report_error(exception, parameters: {event_id: event_id})
+  end
+
+  private
+
+  attr_reader :application
+
+  def build_application(event)
+    app_attrs = {id: event.application.id, user_key: event.user_key, application_id: event.application_id}
+    Cinstance.new(app_attrs, without_protection: true).tap do |app|
```

Comment:
I will refactor in the following PR to use service objects instead of building the models to call to apisonator through them.